contrib/cirrus/lib.sh

```diff
@@ -208,6 +208,20 @@ install_test_configs() {
     # as the default).  This config prevents allocation of network address space used
     # by default in google cloud.  https://cloud.google.com/vpc/docs/vpc#ip-ranges
     install -v -D -m 644 $SCRIPT_BASE/99-do-not-use-google-subnets.conflist /etc/cni/net.d/
+
+    # In Nov. 2020, Docker rate-limits image pulling.  To avoid hitting these
+    # limits while tesing, use the provided google cache.  Integration tests
+    # make use of the copy of this file under `./test/`, however system tests
+    # use the globally installed file.
+    # Ref: https://cloud.google.com/container-registry/docs/pulling-cached-images
+    cat >> ./test/registries.conf <<EOF
+# Added by ${FUNCNAME[0]} in ${BASH_SOURCE[0]}
+[[registry]]
+location="docker.io"
+
+[[registry.mirror]]
+location="mirror.gcr.io"
+EOF
```

Comment:
I prefer to change `./test/registries.conf` directly. This way, we use the same registry everywhere.

Reply:
That's effectively what we get by installing the file system-wide (next line).  I too am not a big fan of having multiple potential locations for the configuration, as it makes maintenance and debugging difficult.  In a perfect world, I'd rather not rely on setting the env. var.  at all.  It's would be simpler if the e2e tests defaulted to using the installed system configuration.  I'm just not knowledgeable enough of their inner-workings to affect that change :disappointed:

Reply:
oh, reading this again, you mean change the file instead of the script?  I was about to do that previously but was concerned that `mirror.gcr.io` was something special for google-cloud VMs.  Sure we can run that way...